pkg/cli/mirror/catalog_images.go

```diff
@@ -223,8 +298,10 @@ func (o *MirrorOptions) processCatalogRefs(ctx context.Context, catalogsByImage
 				containertools.ConfigsLocationLabel: "/configs",
 			}
 			cfg.Config.Labels = labels
-			cfg.Config.Cmd = []string{"serve", "/configs"}
-			cfg.Config.Entrypoint = []string{"/bin/opm"}
+			// Although it was prefered to keep the entrypoint and command as it was
+			// we couldnt guarantie that the cache-dir was /tmp/cache, and therefore
+			// we had to specify the command for the newly built catalog
+			cfg.Config.Cmd = []string{"serve", "/configs", "--cache-dir=/tmp/cache"}
```

Comment:
Still seems like this update of the cmd with --cache-dir needs to be conditional based on if we added a new cache layer. It probably needs to take withCacheRegeneration into account

Reply:
great catch! Thank you!